backend/synthesis/synthesis.py

```diff
@@ -144,6 +145,8 @@ def synthesize(data: dict) -> Tuple[str, BytesIO]:
     zipfile, optional_files = syntheize_modules(data, zipfile)
     zipfile = synthesize_executioner(zipfile, optional_files)
     zipfile = syntesize_extras(zipfile)
+    # Add the .vc3 file to the built application, this will let us easily load the project in Visual Circuit
+    zipfile.append(data['package']['name'] + PROJECT_FILE_EXTENSION, json.dumps(data))
```

Comment:
You might need to handle empty project name, just like the line below this.
Please check once.

Reply:
In that case it becomes a `.vc3` file. No error is thrown... should I add a default name anyways?

Reply:
Ya. I think it's better to have something rather than just extension in the name.